src/DiagnosticFilterer.ts

```diff
@@ -24,7 +28,7 @@ export class DiagnosticFilterer {
         let result = this.getDiagnostics();
 
         //clean up
-        delete this.byFile;
+        this.byFile = {};
```

Comment:
Nowhere checks whether `byFile` is `undefined`, but several places do iterate over its keys:

- `getDiagnostics` iterates over them without checking their presence; this is safe in the current implementation because it runs after `groupByFile` (which assigns a default value of `{}` to `this.byFile`.
- `filterAllFiles` is in the same boat as `getDiagnostics`.
- `filterFile` deletes a field off of `this.byFile` without checking whether `this.byFile` is defined, but this is also safe for the same reasons

Changing the default value to never be undefined prevents excessive checks or assertions in these places.